sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala

```diff
@@ -343,4 +343,23 @@ class JsonExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       null
     )
   }
+
+  test("to_json") {
+    val schema = StructType(StructField("a", IntegerType) :: Nil)
+    val struct = Literal.create(create_row(1), schema)
+    checkEvaluation(
+      StructToJson(Map.empty, struct),
+      """{"a":1}"""
+    )
+  }
+
+  test("to_json - invalid type") {
+    val schema = StructType(StructField("a", CalendarIntervalType) :: Nil)
```

Comment:
I would like to leave a note. In case of CSV we are verifying the types before actually running tasks but for JSON it is not doing this. So, I make this `SparkSQLJsonProcessingException` which is technically a `RuntimeException`. However, if you want me to fix it here (adding a logic to verify the schema ahead) I will definitely do this here together.